drivers/hmis/spec/jobs/hmis/transfer_enrollment_project_job_spec.rb

```diff
@@ -0,0 +1,220 @@
+          source_project_id: source_project.id,
+          target_project_id: 999999,
+        )
+      end.to raise_error('Target project not found: 999999')
+    end
+
+    it 'raises error when projects are in different data sources' do
+      other_data_source = create(:hmis_data_source)
+      other_project = create(:hmis_hud_project, data_source: other_data_source)
+
+      expect do
+        described_class.perform_now(
+          enrollment_ids: [enrollment1.id],
+          source_project_id: source_project.id,
+          target_project_id: other_project.id,
+        )
+      end.to raise_error('Source and target project must be in the same data source')
+    end
+
+    it 'raises error when enrollments not found in source project' do
+      other_enrollment = create(:hmis_hud_enrollment, data_source: data_source)
+
+      expect do
+        described_class.perform_now(
+          enrollment_ids: [enrollment1.id, other_enrollment.id],
+          source_project_id: source_project.id,
+          target_project_id: target_project.id,
+        )
+      end.to raise_error(/Some enrollments not found in source project/)
+    end
+  end
+
+  describe 'household validation' do
+    let!(:household_id) { Hmis::Hud::Base.generate_uuid }
+    let!(:hoh_enrollment) { create(:hmis_hud_enrollment, data_source: data_source, project: source_project, client: client1, household_id: household_id, relationship_to_hoh: 1) }
+    let!(:hhm_enrollment) { create(:hmis_hud_enrollment, data_source: data_source, project: source_project, client: client2, household_id: household_id, relationship_to_hoh: 2) }
+
+    it 'allows transfer when all household members are included' do
+      expect do
+        described_class.perform_now(
+          enrollment_ids: [hoh_enrollment.id, hhm_enrollment.id],
+          source_project_id: source_project.id,
+          target_project_id: target_project.id,
+        )
+      end.not_to raise_error
+
+      hoh_enrollment.reload
+      hhm_enrollment.reload
+      expect(hoh_enrollment.project_pk).to eq(target_project.id)
+      expect(hhm_enrollment.project_pk).to eq(target_project.id)
+    end
+
+    it 'raises error when trying to transfer partial household' do
+      expect do
+        described_class.perform_now(
+          enrollment_ids: [hoh_enrollment.id], # Only transferring HoH, not household member
+          source_project_id: source_project.id,
+          target_project_id: target_project.id,
+        )
+      end.to raise_error(/Cannot transfer partial household/)
+    end
+
+    it 'raises error when trying to transfer only household member without HoH' do
+      expect do
+        described_class.perform_now(
+          enrollment_ids: [hhm_enrollment.id], # Only transferring household member, not HoH
+          source_project_id: source_project.id,
+          target_project_id: target_project.id,
+        )
+      end.to raise_error(/Cannot transfer partial household/)
+    end
```

Comment:
This might not need to be tested as its own case separate from the test on line 184? It didn't seem to me like the partial household validation logic is doing anything specific with HoH.

Reply:
you're right this is unnecessary, removing!